### PR TITLE
domain: set etcdClient loglevel to error

### LINKED ISSUE #15479: Domain: sometimes print warn log when domain closed
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do?
TiDB domain integrated into BR pkg, when BR finished and close the domain sometimes it will print warn log
```
{"level":"warn","ts":"2020-02-27T14:38:56.499+0800","caller":"clientv3/retry_interceptor.go:61","msg":"retrying of unary invoker failed","target":"endpoint://client-9177a4a5-a0a9-4dae-b5a3-83e51c4aada7/127.0.0.1:2379","attempt":0,"error":"rpc error: code = Canceled desc = context canceled"}
```
the reason as I can tell is when dom.Close(), there exists concurrent close, and if etcd session closed first, it will retry sometimes and print above warn log.
it will make user confused whether BR is success or not. please help fix it.

2. What did you expect to see?
no warn log

3. What did you see instead?
sometimes warn log

4. What version of TiDB are you using (`tidb-server -V` or run `select tidb_version();` on TiDB)?
master

### PR DESCRIPTION
**Pull request by 3pointer**
### What problem does this PR solve?

Issue Number: close #15479 

Problem Summary: etcdClient print warn log randomly, when domain closed

### What is changed and how it works?

Proposal: https://github.com/pingcap/tidb/issues/15479#issuecomment-603213430

What's Changed: set etcdClient loglevel to error

How it Works: warn log won't come out

### Related changes

- Need to cherry-pick to the release-3.1

### Check List 

Tests 

- Manual test

### CODE REVIEW

File: domain/domain.go [hunk 1]
```diff
@@ -648,7 +648,11 @@ func (do *Domain) Init(ddlLease time.Duration, sysFactory func(*Domain) (pools.R
 	if ebd, ok := do.store.(tikv.EtcdBackend); ok {
 		if addrs := ebd.EtcdAddrs(); addrs != nil {
 			cfg := config.GetGlobalConfig()
+			// silence etcd warn log
```
Comment (zimulala) [inline domain/domain.go]: It is best to describe in detail the reason for changing to the error level, otherwise, we may forget in the future.

Comment (3pointer) [reply domain/domain.go]: Sure, I add detail  and put issue link in description.

**Comment by zimulala** [comment]
PTAL @tiancaiamao @disksing

**Comment by wjhuang2016** [comment]
Set the etcdClient loglevel to error may increase the difficulty when deal with etcd releated bug, I hope that we can handle it in other way.

**Pull request merged**